### ISSUE TITLE
Build fails with maven-assembly-plugin version 2.5.5

### ISSUE THREAD

**Issue by dpursehouse** [elastic/elasticsearch#11824]
Running the command `mvn install -pl core -DskipTests` on the master branch results in a failure:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 39.880s
[INFO] Finished at: Tue Jun 23 19:50:18 JST 2015
[INFO] Final Memory: 52M/1462M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.5.5:single (default) on project elasticsearch: Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:2.5.5:single failed: group id '2060452353' is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

If I remove the `version` of the maven-assembly-plugin in `pom.xml` like this:

```
$ git diff
diff --git a/pom.xml b/pom.xml
index 6083089..8df98e2 100644
--- a/pom.xml
+++ b/pom.xml
@@ -482,7 +482,6 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-assembly-plugin</artifactId>
-                    <version>2.5.5</version>
                 </plugin>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
```

then it uses version 2.2-beta-5 of the plugin, and the build succeeds.

**Comment by s1monw** [comment]
I just ran it and it works just fine for me maybe it's something else or your version is corrupted?

**Comment by dpursehouse** [comment]
I tried to remove that version from my ~/.m2/repository folder but got the same error after it downloaded again.  And the same error if I explicitly set the version to 2.5.4 or 2.5.3 (the other versions that I happened to have locally).

This is on Ubuntu 12.04 LTS if it matters.

As long as I can build by removing the explicit version in pom.xml it's fine; I was just hoping someone else had also had the same problem and could point me in the right direction to fix it...

**Comment by dadoonet** [comment]
What is your maven version?

**Comment by dpursehouse** [comment]
@dadoonet  version 3.0.4.  Full details below.


```
$ mvn --version
Apache Maven 3.0.4
Maven home: /usr/share/maven
Java version: 1.7.0_45, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-oracle-1.7.0.45/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.2.0-82-generic", arch: "amd64", family: "unix"
```

**Comment by dadoonet** [comment]
@dpursehouse Could you try to update to 3.1.1 and see if it fixes your issue?

**Comment by dpursehouse** [comment]
@dadoonet unfortunately I'm getting the same error also with version 3.1.1.

**Comment by xuzha** [comment]
This also happened to me. The workaround here is to add posix mode tar extensions (tarLongFileMode=posix).  Works for me. 

Maven home: /usr/local/apache-maven/apache-maven-3.3.3
Java version: 1.8.0_45, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_45.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.10.3", arch: "x86_64", family: "mac"

**Comment by dpursehouse** [comment]
@xuzha that has fixed it.  Thanks!

**Comment by xuzha** [comment]
I'm just curious why it doesn't happen to other people, is this related to maven-assembly-plugin upgrade?

**Comment by lmenezes** [comment]
also happened to me, and adding tarLogFileMode also fixed. If this of any interest:

Apache Maven 3.2.5 (12a6b3acb947671f09b81f49094c53f426d8cea1; 2014-12-14T18:29:23+01:00)
Maven home: /usr/local/Cellar/maven/3.2.5/libexec
Java version: 1.8.0_40, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_40.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.10.4", arch: "x86_64", family: "mac"

**Comment by dadoonet** [comment]
This is really weird. I'm running on MacOS as well, with Maven 3.3.3 and never got this issue.
And I'm building elasticsearch from a long pathname: `/Users/dpilato/Documents/Elasticsearch/dev/es-master/elasticsearch`. I would expect getting the same issue as you guys get.

Anyone has an idea on how I could reproduce this? What are the exact commands you are running?
Does it break also when running `mvn clean install -DskipTests` from the elasticsearch root dir?

**Comment by xuzha** [comment]
@dadoonet  I didn't put too much time investigating this, but I couldn't reproduce it now. Looks like this is not related to the long pathname.

**Comment by lmenezes** [comment]
still the same result:
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:41 min
[INFO] Finished at: 2015-09-15T09:33:57+02:00
[INFO] Final Memory: 53M/832M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.5.5:single (default) on project elasticsearch: Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:2.5.5:single failed: group id '1757119603' is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit -> [Help 1]

**Comment by dadoonet** [comment]
@lmenezes could you run the same with `-X` option and upload somewhere the full log?

**Comment by lmenezes** [comment]
@dadoonet let me know if this works: https://www.dropbox.com/s/0o3gfqyxfmaotny/maven.log?dl=0

**Comment by dpursehouse** [comment]
@dadoonet Note that I'm building on Ubuntu, not MacOS.

**Comment by dadoonet** [comment]
Interesting. It sounds like the problem is coming from the `group id` which is set to inner files.

Basically it then fails if for a given tar entry, we have:

```java
    private void failForBigNumbers(TarArchiveEntry entry) {
        this.failForBigNumber("entry size", entry.getSize(), 8589934591L);
        this.failForBigNumber("group id", (long)entry.getGroupId(), 2097151L);
        this.failForBigNumber("last modification time", entry.getModTime().getTime() / 1000L, 8589934591L);
        this.failForBigNumber("user id", (long)entry.getUserId(), 2097151L);
        this.failForBigNumber("mode", (long)entry.getMode(), 2097151L);
        this.failForBigNumber("major device number", (long)entry.getDevMajor(), 2097151L);
        this.failForBigNumber("minor device number", (long)entry.getDevMinor(), 2097151L);
    }
```

I'm trying to get more information now.

**Comment by khmarbaise** [comment]
You should configure the usage of [posix as mentioned in the FAQ.](https://maven.apache.org/plugins/maven-assembly-plugin/faq.html#tarFileModes)

**Comment by dadoonet** [comment]
@khmarbaise sure. Wondering why it's not the default then? Is there any tradeoff?

**Comment by dadoonet** [comment]
As we moved to gradle in master branch, I think we can live with this and default maven options for now. At least we have a workaround if some people get this issue.

So let's close it as "fixed in 3.0" :P

**Comment by larryfast** [comment]
FYI had this problem at our site and the source was userids and groupids. My user & group ID is 831100298 (well, not exactly that).  So the error seems to come from file creation and userids.  And yes, the posix fix worked for me as well.
